src/test/lib/ApiTest.cpp

```diff
@@ -6530,3 +6627,225 @@ QuicTestValidateConnectionPoolCreate()
     }
 }
 #endif // QUIC_API_ENABLE_PREVIEW_FEATURES
+
+void
+QuicTestRetryConfigSetting()
+{
+    uint8_t TestSecret[32]{};
+    uint8_t ResultBuffer[sizeof(QUIC_STATELESS_RETRY_CONFIG) + 32]{};
+    QUIC_STATELESS_RETRY_CONFIG* ResultConfig = (QUIC_STATELESS_RETRY_CONFIG*)ResultBuffer;
+    uint32_t ResultBufferSize = sizeof(ResultBuffer);
+    uint32_t TestRotationMs = 54321;
+    QUIC_AEAD_ALGORITHM_TYPE TestAlg = QUIC_AEAD_ALGORITHM_AES_128_GCM;
+    MsQuicRegistration Registration("TestRetryConfigSetting");
+    {
+        QuicStorageSettingScopeGuard GlobalSettings = QuicStorageSettingScopeGuard::Create();
+
+        //
+        // Test that the defaults were correctly picked up.
+        //
+        TEST_QUIC_SUCCEEDED(
```

Comment:
nit: It'd be nice to test the case where no buffer / a buffer too short is provided, to check we get the length as expected